tests/director/tests/formation_api_test.go

```diff
@@ -116,6 +116,134 @@ func TestApplicationFormationFlow(t *testing.T) {
+	var formation graphql.Formation
+	createReq := fixtures.FixCreateFormationRequest(newFormation)
+	err = testctx.Tc.RunOperation(ctx, certSecuredGraphQLClient, createReq, &formation)
+	require.NoError(t, err)
+	require.Equal(t, newFormation, formation.Name)
+
+	nonExistingFormation := "nonExistingFormation"
+	t.Logf("Shoud not assign runtime to formation %s, as it is not in the label definition", nonExistingFormation)
+	failAssignReq := fixtures.FixAssignFormationRequest(rtm.ID, "RUNTIME", nonExistingFormation)
+	var failAssignFormation *graphql.Formation
+	err = testctx.Tc.RunOperation(ctx, certSecuredGraphQLClient, failAssignReq, failAssignFormation)
+	require.Error(t, err)
+	require.Nil(t, failAssignFormation)
+
+	t.Logf("Assign runtime to formation %s", newFormation)
+	assignReq := fixtures.FixAssignFormationRequest(rtm.ID, "RUNTIME", newFormation)
+	var assignFormation graphql.Formation
+	err = testctx.Tc.RunOperation(ctx, certSecuredGraphQLClient, assignReq, &assignFormation)
+	require.NoError(t, err)
+	require.Equal(t, newFormation, assignFormation.Name)
+
+	t.Log("Check if new scenario label value was set correctly")
+	checkRuntimeFormationLabels(t, ctx, rtm.ID, labelKey, []string{asaFormation, newFormation})
+
+	t.Logf("Assign runtime to formation %s which was already assigned by ASA", asaFormation)
+	assignReq = fixtures.FixAssignFormationRequest(rtm.ID, "RUNTIME", asaFormation)
+	err = testctx.Tc.RunOperation(ctx, certSecuredGraphQLClient, assignReq, &assignFormation)
+	require.NoError(t, err)
+	require.Equal(t, asaFormation, assignFormation.Name)
+
+	t.Log("Check if the formation label value is still assigned")
+	checkRuntimeFormationLabels(t, ctx, rtm.ID, labelKey, []string{asaFormation, newFormation})
+
+	t.Logf("Try to unassign runtime from formation %q which was assigned by ASA", asaFormation)
+	unassignReq := fixtures.FixUnassignFormationRequest(rtm.ID, "RUNTIME", asaFormation)
+	var unassignFormation graphql.Formation
+	err = testctx.Tc.RunOperation(ctx, certSecuredGraphQLClient, unassignReq, &unassignFormation)
+	require.NoError(t, err)
+	require.Equal(t, asaFormation, unassignFormation.Name)
+
+	t.Log("Check that the formation label value is still assigned")
+	checkRuntimeFormationLabels(t, ctx, rtm.ID, labelKey, []string{asaFormation, newFormation})
+
+	t.Log("Should not delete formation while runtime is assigned")
+	deleteRequest := fixtures.FixDeleteFormationRequest(newFormation)
+	var nilFormation *graphql.Formation
+	err = testctx.Tc.RunOperationWithCustomTenant(ctx, certSecuredGraphQLClient, tenantId, deleteRequest, nilFormation)
+	assert.Error(t, err)
+	assert.Nil(t, nilFormation)
+
+	t.Logf("Unassign Runtime from formation %s", newFormation)
+	unassignReq = fixtures.FixUnassignFormationRequest(rtm.ID, "RUNTIME", newFormation)
+	err = testctx.Tc.RunOperation(ctx, certSecuredGraphQLClient, unassignReq, &unassignFormation)
+	require.NoError(t, err)
+	require.Equal(t, newFormation, unassignFormation.Name)
+
```

Comment:
Can we verify that the fomation value is not present in the label after the unassignment here using checkRuntimeFormationLabels